### PR TITLE
Test reports are not being published to Azure Pipelines

### PR DESCRIPTION
**Pull request by vagisha-nidhi**
This PR is related to the issue #5713 
The PR helps to publish test results for reporting in Azure Pipelines.

Azure Pipelines provides excellent reporting capabilities that can be leveraged by this PR.
An example report is shown in the image below.
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "lerna": "lerna",
     "bootstrap": "lerna bootstrap",
     "dev": "lerna run build --stream --parallel",
-    "testonly": "jest",
+    "testonly": "jest --reporters=default --reporters=jest-junit",
     "testall": "npm run testonly -- --coverage --forceExit --runInBand",
```
Comment (timneutkens) [inline package.json]:
```suggestion
    "testall": "npm run testonly -- --coverage --forceExit --runInBand --reporters=default --reporters=jest-junit",
```

File: package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "lerna": "lerna",
     "bootstrap": "lerna bootstrap",
     "dev": "lerna run build --stream --parallel",
-    "testonly": "jest",
+    "testonly": "jest --reporters=default --reporters=jest-junit",
```
Comment (timneutkens) [inline package.json]:
```suggestion
    "testonly": "jest",
```

Comment (timneutkens) [reply package.json]: Move it to `"testall"` so that this doesn't run when locally running tests

Comment (vagisha-nidhi) [reply package.json]: Sure. That makes sense. I have made the change.

File: jest.config.js [hunk 3]
```diff
@@ -8,5 +8,7 @@ module.exports = {
   rootDir: 'test',
   modulePaths: ['<rootDir>/lib'],
   globalSetup: '<rootDir>/jest-global-setup.js',
-  globalTeardown: '<rootDir>/jest-global-teardown.js'
+  globalTeardown: '<rootDir>/jest-global-teardown.js',
+  coverageReporters: ['text', 'cobertura'],
+  collectCoverage: true
```
Comment (timneutkens) [inline jest.config.js]:
```suggestion
```

I think this line can be removed as we pass `--coverage`

Comment (vagisha-nidhi) [reply jest.config.js]: Yes. Removed this.

**Comment by timneutkens** [comment]
@vagisha-nidhi I wonder where the Azure Pipelines check went 🤔

**Comment by vagisha-nidhi** [comment]
@timneutkens I am unable to see Azure Pipelines check on other PRs as well. Can we queue this again for coveralls integration?

**Comment by timneutkens** [comment]
I'm not exactly sure what you mean by queueing this again, but the coveralls integration is triggered here: https://github.com/zeit/next.js/blob/canary/.travis.yml#L24

**Comment by timneutkens** [comment]
The script is here: https://github.com/zeit/next.js/blob/canary/package.json#L15

and it currently fails like this: https://travis-ci.org/zeit/next.js/jobs/457876180#L3606-L3615

**Review by timneutkens — APPROVED** [review]
Great 🙌

**Comment by timneutkens** [comment]
Thank you very much!

**Pull request merged**